stackslib/src/net/download/nakamoto.rs

```diff
@@ -2056,6 +2108,13 @@ impl NakamotoTenureDownloaderSet {
         &mut self,
         tenure_block_ids: &HashMap<NeighborAddress, AvailableTenures>,
     ) {
+        for downloader_opt in self.downloaders.iter_mut() {
+            let Some(downloader) = downloader_opt.as_mut() else {
+                continue;
+            };
+            downloader.try_transition_to_fetch_end_block();
+        }
```

Comment:
Why not just invoke `downloader.try_transition_to_fetch_end_block();` at ~L2134?

Reply:
There are two cases here, and these two methods handle different cases.

* `try_transition_to_fetch_end_block()` handles the case where the downloader has been waiting too long for another downloader to give it its tenure-end block.

* `transition_to_fetch_end_block()` handles the case where the downloader's tenure-end block could _never_ be satisfied by another downloader in the downloading reward cycle (e.g. it's the PoX anchor block tenure and also the last tenure).

I can give them better names if you want.